tests/Console/Commands/EmitEventsTest.php

```diff
@@ -0,0 +1,329 @@
+<?php
+
+namespace Softonic\TransactionalEventPublisher\Console\Commands;
+
+use Closure;
+use Exception;
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use Illuminate\Support\Facades\Log;
+use Mockery;
+use phpmock\mockery\PHPMockery;
+use Softonic\TransactionalEventPublisher\Contracts\EventStoreMiddlewareContract;
+use Softonic\TransactionalEventPublisher\EventStoreMiddlewares\AmqpMiddleware;
+use Softonic\TransactionalEventPublisher\Model\DomainEvent;
+use Softonic\TransactionalEventPublisher\Model\DomainEventsCursor;
+use Softonic\TransactionalEventPublisher\TestCase;
+
+class EmitEventsTest extends TestCase
+{
+    use DatabaseTransactions;
+
+    private readonly EventStoreMiddlewareContract $eventPublisherMiddleware;
+
+    private readonly EmitEvents $emitEvents;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->app->register('Softonic\TransactionalEventPublisher\ServiceProvider');
+
+        $this->loadMigrationsFrom(__DIR__ . '/../../../database/migrations');
+
+        config()->set('transactional-event-publisher.event_publisher_middleware', AmqpMiddleware::class);
+
+        $this->eventPublisherMiddleware = Mockery::mock(EventStoreMiddlewareContract::class);
+        $this->app->instance(EventStoreMiddlewareContract::class, $this->eventPublisherMiddleware);
+
+        $this->emitEvents = new EmitEvents();
+        $this->emitEvents->eventPublisherMiddleware = $this->eventPublisherMiddleware;
+        $this->emitEvents->databaseConnection = 'testing';
+        $this->emitEvents->batchSize = 2;
+    }
+
+    /**
+     * @test
+     */
+    public function whenThereIsNoCursorAndEventsItShouldInitializeItAndCallTheSendBatchesMethod(): void
+    {
+        $emitEvents = Mockery::mock(EmitEvents::class)->makePartial();
+        $emitEvents->__construct();
+        $emitEvents->shouldAllowMockingProtectedMethods();
+        $emitEvents->shouldReceive('sendBatches')->once();
+        $this->app->instance(EmitEvents::class, $emitEvents);
+
+        $this->artisan('event-sourcing:emit');
+
+        self::assertDatabaseHas(DomainEventsCursor::class, ['last_id' => 0]);
+    }
+
+    /**
+     * @test
+     */
+    public function whenThereIsACursorButNoEventsItShouldCallTheSendBatchesMethod(): void
+    {
+        $cursor = DomainEventsCursor::factory()->create();
+
+        $emitEvents = Mockery::mock(EmitEvents::class)->makePartial();
+        $emitEvents->__construct();
+        $emitEvents->shouldAllowMockingProtectedMethods();
+        $emitEvents->shouldReceive('sendBatches')->once();
```

Comment:
Why are not checking that the publish is called once instead of mocking the same object to be tested?

Reply:
I need to mock the sendBatches() method to avoid the infinite loop.